presto-main/src/main/java/com/facebook/presto/metadata/DiscoveryNodeManager.java

```diff
@@ -176,8 +184,15 @@ private void pollWorkers()
 
         // Add new nodes
         for (InternalNode node : aliveNodes) {
-            nodeStates.putIfAbsent(node.getNodeIdentifier(),
-                    new HttpRemoteNodeState(httpClient, uriBuilderFrom(node.getInternalUri()).appendPath("/v1/info/state").build()));
+            if (protocol.equals(CommunicationProtocol.HTTP)) {
```

Comment:
Could we use a switch?

Reply:
We might still need if/else given we will check `node.getThriftPort().isPresent()`. It is possible that user specifies to use thrift but the port is empty due to thrift port in general is not a first-level citizen in discovery service so that the broadcasting could be slower. In that case, we may just treat the node is not ready.

Reply:
Could we do it like

```java
switch(protocol) {
    case THRIFT:
        if (node.getThriftPort().isPresent) {
            // ...
        }
    default:
        throw new Exc...
}
```